packages/manager/src/features/linodes/UserDataAccordion.tsx

```diff
@@ -0,0 +1,94 @@
+import * as React from 'react';
+import Accordion from 'src/components/Accordion';
+import { makeStyles } from 'src/components/core/styles';
+import Typography from 'src/components/core/Typography';
+import ExternalLink from 'src/components/ExternalLink';
+import HelpIcon from 'src/components/HelpIcon';
+import Notice from 'src/components/Notice';
+import TextField from 'src/components/TextField';
+
+interface Props {
+  userData: string | undefined;
+  onChange: (userData: string) => void;
+  disabled?: boolean;
+}
+
+const useStyles = makeStyles(() => ({
+  helpIcon: {
+    padding: '0px 0px 4px 8px',
+    '& svg': {
+      fill: 'currentColor',
+      stroke: 'none',
+    },
+  },
+}));
+
+const UserDataAccordion: React.FC<Props> = (props) => {
+  const { disabled, userData, onChange } = props;
+  const [formatWarning, setFormatWarning] = React.useState(false);
+
+  const classes = useStyles();
+
+  const checkFormat = (userData: string, isOnChange: boolean) => {
+    const userDataLower = userData.toLowerCase();
+    if (
+      userData.length > 0 &&
+      !userDataLower.startsWith('#cloud-config') &&
+      !userDataLower.startsWith('content-type: text/') &&
+      !userDataLower.startsWith('#!')
+    ) {
+      if (!isOnChange) {
+        setFormatWarning(true);
+      }
+    } else {
+      setFormatWarning(false);
+    }
+  };
+
+  const accordionHeading = (
+    <>
+      Add User Data{' '}
+      <HelpIcon
+        text={'Definition of user data and what it is used for'}
+        className={classes.helpIcon}
+        interactive
+      />
+    </>
+  );
+
+  return (
+    <Accordion heading={accordionHeading} style={{ marginTop: 24 }}>
+      <Typography>
+        Paste user data into the field below. Formats accepted are YAML and
+        bash. Helper text with links to docs and guides.
+      </Typography>
+      <TextField
+        label="User Data"
+        multiline
+        rows={1}
+        expand
+        onChange={(e) => {
+          checkFormat(e.target.value, true);
+          onChange(e.target.value);
+        }}
+        value={userData}
+        disabled={Boolean(disabled)}
+        onBlur={(e) => checkFormat(e.target.value, false)}
+        data-qa-user-data-input
+      />
+      {formatWarning ? (
+        <Notice warning important>
```

Comment:
What do you think about adding a little spacing between the bottom of the `TextField` and top of the warning `Notice`, or getting some feedback from UX on their thoughts? 

**Now**:
![Screen Shot 2023-02-27 at 7 49 35 AM](https://user-images.githubusercontent.com/114685994/221616040-bcdaa01a-fe8e-46ff-b7af-b84a9a4a1976.jpg)


**With some margin** (this was a top margin of 24 to match default bottom margin styling for `Notice`, but may be too much space):

```suggestion
        <Notice warning important spacingTop={24}>
```
![Screen Shot 2023-02-27 at 7 59 51 AM](https://user-images.githubusercontent.com/114685994/221616062-2676a2fa-8aa5-433c-8e8a-5a8662ae3dc2.jpg)